### PR TITLE
ci/ui: add UI upgrade test back

### LINKED ISSUE #659: e2e UI: Test upgrade in a dedicated test
Follow up of https://github.com/rancher/elemental/issues/651

Until now, we test all the thing in one test and that's not the proper way to test upgrade.
For testing upgrade, we have to install from stable and then upgrade to dev/staging.

We will probably use this https://github.com/rancher/elemental/issues/649

That means we need to switch upgrade from our nightly test and move it to a second nightly test.

- [x] Disable upgrade in our main scenario https://github.com/rancher/elemental/pull/679
- [x] Research a way to tell Cypress which tests to run https://github.com/rancher/elemental/issues/680
- Add the new upgrade tests:
- [ ] Upgrade with osImage https://github.com/rancher/elemental/pull/683
- [ ] Upgrade by using OS Channels (not ready yet in the UI)

### PR DESCRIPTION
**Pull request by juadk**
Fix #659 

Verification runs:
[UI-K3s-Stable-RM_Latest-Upgrade](https://github.com/rancher/elemental/actions/runs/4247873025/jobs/7386410896) - This one failed for a reason external to the PR.
[UI-K3s-Stable-RM_Stable-Upgrade](https://github.com/rancher/elemental/actions/runs/4247873032/jobs/7386411012) :heavy_check_mark: 
[UI-RKE2-Stable-RM_Latest-Upgrade](https://github.com/rancher/elemental/actions/runs/4247873037/jobs/7386409520):heavy_check_mark: 
[UI-RKE2-Stable-RM_Stable-Upgrade](https://github.com/rancher/elemental/actions/runs/4247873028/jobs/7386412170):heavy_check_mark: 

- cluster renamed from `myelementalcluster` to `mycluster` due to this bug https://github.com/rancher/elemental-operator/issues/365
- update the regex which is making sure we went through proxy

### CODE REVIEW

**Comment by juadk** [comment]
In the meantime, I'm trying to check if we can remove the workaround we have for upgrade (restart fleet controller and fleet agent).
First test without it ok: https://github.com/rancher/elemental/actions/runs/4233167479
But as it was sporadic, I need to trigger more runs to validate.

**Comment by juadk** [comment]
Finally, without the workaround, it's still failing sometimes so I will keep it for now.
I will continue my investigations later.

File: .github/workflows/ui-k3s-os-upgrade.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      destroy_runner: ${{ inputs.destroy_runner }}
+      elemental_ui_version: ${{ inputs.elemental_ui_version }}
+      iso_to_test: ${{ inputs.iso_to_test }}
+      k8s_version_to_provision: v1.24.8+k3s1
+      proxy: ${{ inputs.proxy }}
+      rancher_channel: ${{ inputs.rancher_channel }}
+      rancher_version: ${{ inputs.rancher_version }}
+      runner_template: ${{ inputs.runner_template }}
+      test_type: ui
+      upgrade_image: registry.opensuse.org/isv/rancher/elemental/${{ inputs.upgrade_os_channel }}/teal53/15.4/rancher/elemental-teal/5.3:latest
```
Comment (ldevulder) [inline .github/workflows/ui-k3s-os-upgrade.yaml:59]: `upgrade_os_channel` is not defined in the `inputs`.

File: .github/workflows/ui-rke2-os-upgrade.yaml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      destroy_runner: ${{ inputs.destroy_runner }}
+      elemental_ui_version: ${{ inputs.elemental_ui_version }}
+      iso_to_test: ${{ inputs.iso_to_test }}
+      k8s_version_to_provision: v1.24.8+rke2r1
+      proxy: ${{ inputs.proxy }}
+      rancher_channel: ${{ inputs.rancher_channel }}
+      rancher_version: ${{ inputs.rancher_version }}
+      runner_template: ${{ inputs.runner_template }}
+      test_type: ui
+      upgrade_image: registry.opensuse.org/isv/rancher/elemental/${{ inputs.upgrade_os_channel }}/teal53/15.4/rancher/elemental-teal/5.3:latest
```
Comment (ldevulder) [inline .github/workflows/ui-rke2-os-upgrade.yaml:59]: `upgrade_os_channel` is not defined in the `inputs`.

**Review by ldevulder — CHANGES_REQUESTED** [review]
2 small fixes to do.

**Review by ldevulder — APPROVED** [review]

**Pull request merged**